components/br-stream/src/observer.rs

```diff
@@ -72,5 +150,160 @@ impl<E: KvEngine> CmdObserver<E> for BackupStreamObserver {
         }
     }
 
-    fn on_applied_current_term(&self, _: StateRole, _: &Region) {}
+    fn on_applied_current_term(&self, role: StateRole, region: &Region) {
```

Comment:
it looks like only leader will call back this function. https://github.com/tikv/tikv/blob/c778d6ec2224c43b2a57a35e497cd7c63ecb2b91/components/raftstore/src/store/peer.rs#L2715-L2718.
so we can remove `if role == StateRole::Leader`. but why use this callback?

Reply:
because the resolved TS observer did so.
https://github.com/3pointer/tikv/blob/52406d9df82d9a31c619368add5efe34ec67895b/components/resolved_ts/src/observer.rs#L74-L83